### ISSUE TITLE
GFA2: grammar for end positions is unclear

### ISSUE THREAD

**Issue by kortschak** [GFA-spec/GFA-spec#81]
The production for positions, `<pos> <- <int>{$}` indicates that all positions have an optional "$" sentinel. However, the text says "A $-sign must follow an integer x if and only if it is the last position in the segment it refers to, i.e. x = L. It is an error to do otherwise.".

This suggests to me that the edge and fragment productions should be rewritten as
```
<fragment> <- F <sid:id> <external:ref>
                  <sbeg:int> <send:int> <fbeg:int> <fend:int> <alignment> <tag>*

<edge>     <- E <eid:opt_id> <sid1:ref> <sid2:ref>
                          <beg1:int> <end1:int>{$} <beg2:int> <end2:int>{$} <alignment> <tag>*
```
(assuming negative integers are valid for fragments) and the `<pos>` production should be dropped.

**Comment by kortschak** [comment]
Also note that the yacc grammar doesn't agree with the current spec grammar in this regard. It has no notion of a sentinel.

**Comment by sjackman** [comment]
The `$` sentinel is handled by terminal `POSITION` token in the Flex grammar:
https://github.com/sjackman/gfalint/blob/master/gfal.l#L36

**Comment by kortschak** [comment]
Ah, yes. Thanks.

**Comment by sjackman** [comment]
> Without examples, the text and grammar are significantly at odds and do not help clarify the situation.

I don't believe the grammar and the text contradict each other. For a file to be valid, it most be both syntactically correct as well as semantically correct. For example, a file with a duplicate segment ID is syntactically correct but not semantically correct. Similarly a record with an incorrect `$` sentinel is syntactically correct but not semantically correct.

I have closed this issue because I believe the spec is correct is written. It would however be improved by an example of a blunt edge. Feel free to open a PR to add an example once you feel that you understand the issue. I'm happy to discuss the issue in the mean time.

> Given that I clearly don't understand the intent of what is written, I would think a PR would be fool hardy. 

A work-in-progress PR can be a great way to discuss a concrete solution to an issue.

**Comment by kortschak** [comment]
OK. That's clear now. I think it may be clarified easily in test with the addition like so, "Positions are conceptually tick-marks between symbols starting a 0 to the left of the first symbol and ending at L to the right of the last symbol where L is the*, possibly zero,* length of the segment." at the cost of style. I think my thinking was contaminated by SAM notions where 0 is anathema.

I can put together a PR at some time for an example, but it would be nice to know what kind of change is expected (and if graphics are wanted where the source for the figures lives).

**Comment by sjackman** [comment]
A simple text example of an abut edge would be a good starting point. A figure would be a bonus.
